src/main/java/org/cryptomator/cryptofs/CryptoFileSystems.java

```diff
@@ -54,9 +54,11 @@ public CryptoFileSystemImpl create(CryptoFileSystemProvider provider, Path pathT
 			rawKey = key.getEncoded();
 			var config = configLoader.verify(rawKey, Constants.VAULT_VERSION);
 			var adjustedProperties = adjustForCapabilities(pathToVault, properties);
+			Cryptor cryptor = config.getCipherCombo().getCryptorProvider(csprng).withKey(key);
```

Comment:
I wonder, how the masterkey survives this method... It will be destroyed when leavin the try-with-resource block. Thus the cryptor should no longer be useful... 🤔

Reply:
Will be discussed in https://github.com/cryptomator/cryptolib/issues/19